File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/livedata/javacode/BaseLifecycleActivity.java
package com.ponthaitay.lifecycleactivity.googleio2017.livedata.javacode;

import android.arch.lifecycle.LifecycleRegistry;
import android.arch.lifecycle.LifecycleRegistryOwner;
import android.support.v7.app.AppCompatActivity;

public class BaseLifecycleActivity extends AppCompatActivity implements LifecycleRegistryOwner {

    private final LifecycleRegistry mRegistry = new LifecycleRegistry(this);

    public BaseLifecycleActivity() {
    }

    @Override
    public LifecycleRegistry getLifecycle() {
        return mRegistry;
    }
}

File: app/src/main/java/com/ponthaitay/lifecycleactivity/googleio2017/livedata/kotlincode/BaseLifecycleActivity.kt
package com.ponthaitay.lifecycleactivity.googleio2017.livedata.kotlincode

import android.arch.lifecycle.LifecycleRegistry
import android.arch.lifecycle.LifecycleRegistryOwner
import android.support.v7.app.AppCompatActivity

open class BaseLifecycleActivity : AppCompatActivity(), LifecycleRegistryOwner {

    @Suppress("LeakingThis")
    private val mRegistry = LifecycleRegistry(this)

    override fun getLifecycle(): LifecycleRegistry  = mRegistry
}